aten/src/ATen/native/quantized/cpu/tensor_operators.cpp

```diff
@@ -0,0 +1,54 @@
+#include <ATen/ATen.h>
+#include <ATen/ExpandUtils.h>
+#include <ATen/NativeFunctions.h>
+
+namespace at {
+namespace native {
+
+/*
+All comparator operators will be named "<aten op name>_quantized_cpu".
+'_out' will be appended for the 'out' variant of the op.
+*/
+
+#define DEFINE_COMPARATOR(at_op) \
+Tensor& at_op##_out_quantized_cpu(Tensor& out, const Tensor& self, \
+                                Scalar other) { \
+  TORCH_CHECK(out.dtype() == at::ScalarType::Bool, \
+              "The 'out' tensor must have dtype 'torch.bool'"); \
+  const auto& self_dq = self.dequantize(); \
```

Comment:
Why are all the results of dequantize() const references instead of values? Technically it's correct due to rvalue lifetime extension but it's confusing to the reader, especially in the presence of `auto`

Reply:
I got used to having const ref whenever I use `auto` keyword -- this was an enforced rule in my previous life. Is it an anti-pattern ?